linalg/simd/m128.hpp

```diff
@@ -14,13 +14,6 @@
 
 #ifdef __SSE2__
 
-#include "../../config/tconfig.hpp"
-#if defined(__x86_64__)
-#include <x86intrin.h>
-#else // assuming MSVC with _M_X64 or _M_IX86
-#include <intrin.h>
-#endif
-
```

Comment:
I wanted to have these lines in each of the files to make sure the header is self-contained and I get proper syntax parsing in qtctreator. To avoid the duplication, we can create small separate headers for this, e.g. `linalg/simd/_x86.hpp` or `linalg/simd/x86_common.hpp` in this case.